core/web/resolver/p2p_test.go

```diff
@@ -0,0 +1,204 @@
+package resolver
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/smartcontractkit/chainlink/core/services/keystore"
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/p2pkey"
+)
+
+func TestResolver_GetP2PKeys(t *testing.T) {
+	t.Parallel()
+
+	query := `
+		query GetP2PKeys {
+			p2pKeys {
+				results {
+					id
+					peerId
+					publicKey
+				}
+			}
+		}
+	`
+
+	fakeKeys := []p2pkey.KeyV2{}
+	expectedKeys := []map[string]string{}
+	for i := 0; i < 2; i++ {
+		k, err := p2pkey.NewV2()
+		assert.NoError(t, err)
+		fakeKeys = append(fakeKeys, k)
+		expectedKeys = append(expectedKeys, map[string]string{
+			"id":        k.ID(),
+			"peerId":    k.PeerID().String(),
+			"publicKey": k.PublicKeyHex(),
+		})
+	}
+
+	d, err := json.Marshal(map[string]interface{}{
+		"p2pKeys": map[string]interface{}{
+			"results": expectedKeys,
+		},
+	})
+	assert.NoError(t, err)
+	expected := string(d)
+
+	testCases := []GQLTestCase{
+		unauthorizedTestCase(GQLTestCase{query: query}, "p2pKeys"),
+		{
+			name:          "success",
+			authenticated: true,
+			before: func(f *gqlTestFramework) {
+				f.Mocks.p2p.On("GetAll").Return(fakeKeys, nil)
+				f.Mocks.keystore.On("P2P").Return(f.Mocks.p2p)
+				f.App.On("GetKeyStore").Return(f.Mocks.keystore)
+			},
+			query:  query,
+			result: expected,
+		},
+	}
+
+	RunGQLTests(t, testCases)
+}
+
+func TestResolver_CreateP2PKey(t *testing.T) {
+	t.Parallel()
+
+	query := `
+		mutation CreateP2PKey {
+			createP2PKey {
+				key {
+					id
+					peerId
+					publicKey
+				}
+			}
+		}
+	`
+
+	fakeKey, err := p2pkey.NewV2()
```

Comment:
maybe use use `cltest.DefaultP2PKey` here

Reply:
I'd prefer this way as I would like to start moving away from depending on `cltest`. It is a massive utility package which does too much